geometries/ramangle/ramangle_algorithm.py

```diff
@@ -1,77 +1,42 @@
-import math as m
 import spiceypy as sp
 
-def ramangle(utc, target, galaxy_targ):
-    """Calculates ram angle, which is measured between velocity vector of Aspera & vector from Aspera
-    to galaxy.
+def ramangle(utc, instr):
+    """Calculates ram angle, which is measured between Earth-relative
+    velocity vector of Aspera & boresight vector of instrument
 
     Args:
-        mkfile (str): metakernel containg data on Aspera, nearby bodies, and M82
         utc (str): date and time at which altitude will be found
-        galaxy_targ (str): body ID for galaxy contained in mkfile
+        instr (str): instrument from which to extract frame & boresight
 
     Returns:
         float: ram angle
     """
 
-    # # # # # PART 1: ANGLE BETWEEN VELOCITY AND POSITION VECTORS # # # # #
+    # # # # # PART 1: ANGLE BETWEEN EARTH-RELATIVE VELOCITY AND BORESIGHT VECTORS # # # # #
 
     et = sp.str2et(utc)
-    ref = 'J2000'
     abcorr = 'NONE'
 
+    #btc get instrument frame and borsight, also instrument frame center
+    ref,vbore = sp.getfov(sp.bods2c(instr),99,99,99)[1:3]
+    target = sp.bodc2s(sp.frinfo(sp.namfrm(ref))[0])
+
     # Generate velocity vector of Aspera wrt Earth
+    #btc ... in instrument frame
     aspera_state, lt = sp.spkezr(target, et, ref, abcorr, 'EARTH')
     vel = aspera_state[3:]
 
-    # Generate position vector of galaxy wrt Aspera
-    galaxy_state, lt = sp.spkezr(galaxy_targ, et, ref, abcorr, target)
-    pos = galaxy_state[0:3]
+    #btc galaxy target is irrelevant
 
-    # Find angle between velocity and position vectors, convert from rad to deg
-    ram_rad = sp.vsep(vel, pos)
+    # Find angle between velocity and bore vectors, convert from rad to deg
+    #btc velocity and boresight vectors are already in instrument frame
+    ram_rad = sp.vsep(vel, vbore)
     ram_deg = sp.convrt(ram_rad, 'RADIANS', 'DEGREES')
 
-    # # # # # PART 2: S/C VELOCITY COMPONENT WRT TARGET # # # # #
-
-    vel_sc_los = sp.vnorm(vel) * m.cos(ram_rad)
-    return ram_deg, vel_sc_los
+    # # # # # PART 2: S/C VELOCITY COMPONENT ALONG BORESIGHT # # # # #
+    vel_sc_los = sp.vdot(sp.vhat(vbore), vel)
```

Comment:
We can get rid of this, as there is a separate routine to calculate S/C veclocity along boresight